doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+
+8.  As a client I want to connect and interact with Databroker either if it contains data or not.
+
+10.  As a client I want to know if there is a provider up and running by getting some sensor or actuator value.
+
+11.  As a client I don’t want to `Subscribe` to target values.
+
+12.  As a client I don’t want to `Subscribe` to attributes values.
+
+13.  As a client I don’t want to `Get` target values.
+
+14.  As a client I don’t want to `Set` sensor values.
+
+15.  As a client I don’t want to `Set` attributes values.
+
+### As Databroker
+1. The data broker shall be able to handle catalogs of signals described by the syntax as defined by the COVESA Vehicle Signal Specification (VSS). This relates to all aspects of the VSS syntax definition, which is also called VSS rule set. This implies that the data broker can handle the signal catalog as defined by the COVESA VSS.
+
+2. Databroker shall support at least those metadata elements as defined by the VSS rule set.
+
+### As Provider
+1. As a provider I want to `Set` the value and current value of one or many sensors and/or actuators with its timestamp to Databroker.
+2. 
+3. A provider shall be able to continuously and efficiently (e.g. via grpc stream) _**update**_ the current value of one or multiple sensors and/or actuators with its timestamp.
+
+4. A provider shall be able to `Subscribe` to the databroker to receive actuation requests.
+
+5. As a provider I want to `Set` once attributes values.
+
+6. As a provider I don’t need to but I would like to `Get` sensor values or current values of actuatros?.
+   
+7. As a provider I don’t want to `Set` target values to Databroker.
+
+# Non-Functional Requirements
+
+## Data availability/persistence according to lifecycle of Client, Databroker and Provider
+1. The data broker either ...
+   * Stores last set values during its own lifecycle,
+   * ? stores values during the system's power cycle (i.e., "persists" values over own restarts, or
+   * store values over system's power cycles.
+
+2. How to "reset" values availability if its provider got inactive (without reseting the value)?
+   * -> Client's job (e.g. using timestamp)?
+   * -> Broker's job (e.g. using timestamp + minimal update cycle)?
+
+3. Provider and client aliveness
+   * If there is no active client subscription should the provider stop sending values to Databroker?
+   * If there is no active provider setting values while client subscription? Should Databroker or Client be aware of it?
+
+## Wildcard support
+* Should `Get` and `Subscribe` calls support Wildcard request?
+* Do we want it only for `GetMetadata`?
+
+Reference -> [Wildcard](./wildcard_matching.md)
+
+## Registration of Datapoints
+Do we need a method for providers to register data points at runtime? **Implemented in old API?**:
+
+Its purpose would be:
+1.	Publishing the availability of new data points during runtime
```

Comment:
What kind of data point do you expect to show up during runtime which are not known at design time/configuration time of the databroker? 

To me there is a difference between signals which are currently not available or that will not be available at all. With the VSS tree I would express which signals can become available in that broker even if they might not be available at the moment.

Reply:
You could think of a scenario, where you add new applications and providers via an app store. With that, new signals could become available during runtime. This could be the case if providing hardware is present but software is missing (during delivery of the vehicle). Or "higher value" signals (based on pure software) could be offered this way. User then can download by payment. 

We could handle this via updating the local broker config and then restarting the broker (loading the new config).

Reply:
At least in the hardware example, I would expect that the tree already contains the signals that the providing hardware could provide since the hardware is then known at design time of the system and hardware. 

That was at least my assumption to differentiate between signals that are currently not available (hardware present but provider missing) and that will not be available at all. 

In general, to have the option to add (or remove?) signals during runtime seems to add more complexity and edge cases which is another reason why I like the approach of updating the local broker config more.

Reply:
I disagree putting signal definitions into the config just because the providing hardware is present. I'd prefer to have only that signals in the config, that have a installed (and runnable) provider in the system. "Currently not available" means for me that the respective provider is currently not up and running (i.e. not yet started or temporary inactive due to a crash or update), or that some other component down the chain to the hardware is temporary not active.